src/app/dashboard/Missions/MissionEditView.jsx

```diff
@@ -246,6 +264,31 @@ const MissionEditView = ({ mission, toDetailsView, toListView }) => {
       deliveryWindow: {
         startTime: delivery.toString(),
       },
+      volunteerUid:
+        selectedVolunteer && values.status !== MissionStatus.tentative
+          ? selectedVolunteer.volunteerUid
+          : "",
+      volunteerDisplayname:
+        selectedVolunteer && values.status !== MissionStatus.tentative
+          ? selectedVolunteer.volunteerDisplayName
+          : "",
+      volunteerPhoneNumber:
+        selectedVolunteer && values.status !== MissionStatus.tentative
+          ? selectedVolunteer.volunteerPhoneNumber
+          : "",
+      tentativeVolunteerUid:
+        selectedVolunteer && values.status === MissionStatus.tentative
+          ? selectedVolunteer.volunteerUid
+          : "",
+      tentativeVolunteerDisplayName:
+        selectedVolunteer && values.status === MissionStatus.tentative
+          ? selectedVolunteer.volunteerDisplayName
+          : "",
+      tentativeVolunteerPhoneNumber:
+        selectedVolunteer && values.status === MissionStatus.tentative
+          ? selectedVolunteer.volunteerPhoneNumber
+          : "",
+      readyToStart: values.readyToStart,
```

Comment:
```
const getVolunteerAttribute = (selectedVolunteer, status, attr) => {
    if(selectedVolunteer && status === MissionStatus.tentative){
        return selectedVolunteer[attr]
   }
   return ""
}

{...,
tentativeVolunteerUid = getVolunteerAttribute(selectedVolunteer, values.status, "volunteerUid")
}
```

Reply:
Had to add one more param to the "getVolunteerAttribute" function but should capture what you are pointing out.

Reply:
It did not look as pretty as I had hope, my bad there, lets keep it for now